apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -354,14 +359,17 @@ export class ChangeAction extends BaseRushAction {
     });
   }
 
-  private _hasProjectChanged(changedFolders: (string | undefined)[], projectFolder: string): boolean {
-    for (const folder of changedFolders) {
-      if (folder && Path.isUnderOrEqual(folder, projectFolder)) {
-        return true;
-      }
+  private _hasProjectChanged(changedFiles: string[], projectFolder: string): boolean {
+    const relevantFiles: string[] = changedFiles.filter((file) => Path.isUnderOrEqual(file, projectFolder));
+
+    const changeIgnoreFile: string = path.join(projectFolder, CHANGE_IGNORE_FILE_NAME);
+
+    const ignoreConfig: Ignore = ignore();
+    if (relevantFiles.length > 0 && FileSystem.exists(changeIgnoreFile)) {
```

Comment:
The `> 0` check is an optimisation: it avoids accessing the filesystem when there are no changes for this project (for most changes in large repos, this will be the case for almost every project). In that case, the `.some(...)` call below instantly returns false anyway.